icicle/tests/test_hash_api.cpp

```diff
@@ -561,200 +561,206 @@ TEST_F(HashApiTest, MerkleTreeBasic)
-
-  ICICLE_LOG_DEBUG << "19 hashes (Total hashes in layer 0 - 1) - full";
-  test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves - nof_leaves_in_hash, leaves);
-  ICICLE_LOG_DEBUG << "19 hashes (Total hashes in layer 0 - 1) - last hash not full";
-  test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves - nof_leaves_in_hash - 1, leaves);
-
-  ICICLE_LOG_DEBUG << "16 hashes (Batch size) - full";
-  test_merkle_tree(hashes, config, output_store_min_layer, 16 * nof_leaves_in_hash, leaves);
-  ICICLE_LOG_DEBUG << "16 hashes (Batch size) - last hash not full";
-  test_merkle_tree(hashes, config, output_store_min_layer, 16 * nof_leaves_in_hash - 1, leaves);
-  ICICLE_LOG_DEBUG << "17 hashes (Batch size + 1) - full";
-  test_merkle_tree(hashes, config, output_store_min_layer, 17 * nof_leaves_in_hash, leaves);
-  ICICLE_LOG_DEBUG << "17 hashes (Batch size + 1) - last hash not full";
-  test_merkle_tree(hashes, config, output_store_min_layer, 17 * nof_leaves_in_hash - 1, leaves);
-
-  ICICLE_LOG_DEBUG << "1 hash - full";
-  test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves_in_hash, leaves);
-  ICICLE_LOG_DEBUG << "1 leaf in tree";
-  test_merkle_tree(hashes, config, output_store_min_layer, 1, leaves);
-
-  ICICLE_LOG_DEBUG << "A whole number of hashes is missing";
-  int nof_hashes = ((rand() % (total_nof_input_hashes - 2)) + 1);
-  ICICLE_LOG_DEBUG << "Number of used hashes: " << nof_hashes << " / " << total_nof_input_hashes;
-  test_merkle_tree(hashes, config, output_store_min_layer, nof_hashes * nof_leaves_in_hash, leaves);
-
-  ICICLE_LOG_DEBUG << "Random amount of leaves";
-  int nof_partial_leaves = ((rand() % nof_leaves) + 1);
-  ICICLE_LOG_DEBUG << "Random amount of leaves: " << nof_partial_leaves << " / " << nof_leaves;
-  test_merkle_tree(hashes, config, output_store_min_layer, nof_partial_leaves, leaves);
-
-  ICICLE_LOG_DEBUG << "Last leaf isn't fully occupied - check that build should fail";
-  auto byte_leaves = reinterpret_cast<const std::byte*>(leaves);
-  int byte_size;
-  do {
-    byte_size = rand() % (nof_leaves * leaf_size);
-  } while (byte_size % leaf_size == 0);
-  byte_size = 327;
-  ICICLE_LOG_DEBUG << "Size of input in bytes: " << byte_size << "\t(" << float(byte_size) / leaf_size << " / "
-                   << nof_leaves << " leaves)";
+    auto config = default_merkle_tree_config();
+    // Test zero padding
+    config.padding_policy = PaddingPolicy::LastValue;
+
+    ICICLE_LOG_VERBOSE << "Full tree";
+    test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves, leaves);
+
+    ICICLE_LOG_VERBOSE << "Last hash isn't full";
+    test_merkle_tree(hashes, config, output_store_min_layer, nof_leaves - 1, leaves);
+
+    const unsigned nof_leaves_in_hash = layer0_hash.default_input_chunk_size() / leaf_size;
+
+    ICICLE_LOG_VERBOSE << "19 hashes (Total hashes in layer 0 - 1) - full";
```

Comment:
Same here - remove 19 and keep the text in the parenthesis.

Reply:
same